src/gui/qgsmapcanvas.cpp

```diff
@@ -202,6 +202,7 @@ QgsMapCanvas::QgsMapCanvas( QWidget * parent, const char *name )
   mCurrentLayer = NULL;
   mMapOverview = NULL;
   mMapTool = NULL;
+  mOldCursor = cursor();
```

Comment:
If the cursor is changed within the map tool it is the map tools responsibility to restore it in the end.

Is there a reason you would change the cursor in a slot connected to this signal. Actually, this could happen in any slot connected to any signal. How is this particular signal special?

I wonder if we could just remove this line instead.